api/src/main/java/org/apache/iceberg/util/StructProjection.java

```diff
@@ -58,6 +58,19 @@ public static StructProjection create(Schema dataSchema, Schema projectedSchema)
     return new StructProjection(dataSchema.asStruct(), projectedSchema.asStruct());
   }
 
+  /**
+   * Creates a projecting wrapper for {@link StructLike} rows.
+   * <p>
+   * This projection does not work with repeated types like lists and maps.
+   *
+   * @param structType type of rows wrapped by this projection
+   * @param projectedStructType result type of the projected rows
+   * @return a wrapper to project rows
+   */
+  public static StructProjection create(StructType structType, StructType projectedStructType) {
+    return new StructProjection(structType, projectedStructType, false);
```

Comment:
Can this use the constructor that defaults `false` instead of adding a default here? That way refactoring later is easier.

Reply:
Fixed.